tests/lib/rules/no-whitespace-before-property.js

```diff
@@ -47,16 +61,25 @@ ruleTester.run("no-whitespace-before-property", rule, {
         "foo\n.bar()\n.baz()",
         "foo.\nbar().\nbaz()",
         "foo\n.bar\n[baz]",
+        "foo\n.bar\n['baz']",
+        "foo\n.bar\n[0]",
+        "foo\n.bar\n[ baz ]",
         "foo\n .bar\n .baz",
         "foo.\n bar.\n baz",
         "foo\n .bar()\n .baz()",
         "foo.\n bar().\n baz()",
         "foo\n .bar\n [baz]",
+        "foo\n .bar\n ['baz']",
+        "foo\n .bar\n [0]",
+        "foo\n .bar\n [ baz ]",
         "foo\n\t.bar\n\t.baz",
         "foo.\n\tbar.\n\tbaz",
         "foo\n\t.bar()\n\t.baz()",
         "foo.\n\tbar().\n\tbaz()",
-        "foo\n\t.bar\n\t[baz]"
+        "foo\n\t.bar\n\t[baz]",
+        "foo\n\t.bar\n\t['baz']",
+        "foo\n\t.bar\n\t[0]",
+        "foo\n\t.bar\n\t[ baz ]"
```

Comment:
I'd like to see tests which include parenthesised targets and parenthesised properties with various spacing inside/outside the parentheses.